data/src/main/java/com/open/data/network/NetworkModule.kt

```diff
@@ -42,7 +51,7 @@ object NetworkModule {
 
     @Provides
     @Singleton
-    fun provideWeatherRepository(apiService: WeatherApiService): WeatherRepository {
-        return WeatherRepository(apiService)
+    fun provideWeatherRepository(apiService: WeatherApiService): WeatherRepositoryImpl {
```

Comment:
DI never returns an implementation (`Impl`) class. It always returns the parent Interface/Abstract class. Hence, this function should return `WeatherRepository` where in its body, it should return an instance of `WeatherRepositoryImpl`.